index.js

```diff
@@ -228,92 +228,78 @@ function inferTypeByKeyword (schema) {
   return schema.type
 }
 
-function addPatternProperties (location) {
+function buildExtraObjectPropertiesSerializer (location) {
   const schema = location.schema
-  const pp = schema.patternProperties
+  const propertiesKeys = Object.keys(schema.properties || {})
```

Comment:
at this point we have the length of properties. What if it is empty? So could we skip stuff?

Reply:
No, the Keys properties are needed here in order not to add the same property twice. If the property was already serialized as a regular property, we should not serialize it as a pattern or additional property. So if the user passes an empty `property` value, the propertiesKeys would be an empty array, and all other values would be added as pattern or additional values.